clang/lib/Headers/hlsl/hlsl_detail.h

```diff
@@ -45,6 +45,10 @@ template <typename T> struct is_arithmetic {
   static const bool Value = __is_arithmetic(T);
 };
 
+template <typename T, int N>
+using HLSL_FIXED_VECTOR =
+    vector<__detail::enable_if_t<(N > 1 && N <= 4), T>, N>;
```

Comment:
should the enable_if_t not go around the entire vector<T,N>, instead of within the T part?

Reply:
i tried that, but if you do `__detail::enable_if_t<(N > 1 && N <= 4),vector<T,N>>;` the N template argument gets disambiguated.  `note: candidate template ignored: couldn't infer template argument 'N'` https://godbolt.org/z/8a1sKWrcW